fix(curriculum): Convert blockquote elements to triple backtick syntax for JavaScript Algorithms and Data Structures

RandellDawson opened the pull request:
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

**PR Purpose:** Converts `blockquote` elements to triple backtick syntax in the `Description` and `Instructions` sections of the `JavaScript Algorithms and Data Structures` challenges.

Related to issue #35991

RandellDawson committed:
fix: convert js algorithms and data structures

RandellDawson committed:
fix: revert some blocks back to blockquote

RandellDawson committed:
fix: reverted comparison code block to blockquotes

moT01 reviewed (approved):
Looked at each of these locally - they all look good to me 👍 nice job @RandellDawson! 🎉

moT01 commented:
es6/understand-the-differences-between-import-and-require.english.md - I have a PR open on this file
es6/write-higher-order-arrow-functions.english.md - you have an open PR on this file

There's a third PR of mine that already got the backtick conversion - so there shouldn't be a problem there.

I'm not sure if you want to remove those files from this PR - or just merge it as is and fix any conflicts that might arise - I'm fine with merging this ahead of mine, I can fix conflicts there.

RandellDawson commented:
I think we will just fix any that pop up.

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-equality-operator.english.md:
````diff
@@ -9,26 +9,10 @@ videoUrl: 'https://scrimba.com/c/cKyVMAL'
-}
-```
-
+<blockquote>function equalityTest(myVal) {<br>&nbsp;&nbsp;if (myVal == 10) {<br>&nbsp;&nbsp;&nbsp;&nbsp; return "Equal";<br>&nbsp;&nbsp;}<br>&nbsp;&nbsp;return "Not Equal";<br>}</blockquote>
````
Shouldn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-equality-operator.english.md:
````diff
@@ -9,26 +9,10 @@ videoUrl: 'https://scrimba.com/c/cKyVMAL'
-"3"  ==  3    // true
-```
-
+<blockquote>1   ==  1    // true<br>1   ==  2    // false<br>1   == '1'   // true<br>"3"  ==  3    // true</blockquote>
````
Shouldn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-inequality-operator.english.md:
````diff
@@ -9,15 +9,7 @@ videoUrl: 'https://scrimba.com/c/cdBm9Sr'
-0 != false  // false
-```
-
+<blockquote>1 != 2      // true<br>1 != "1"    // false<br>1 != '1'    // false<br>1 != true   // false<br>0 != false  // false</blockquote>
````
Shouldn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-less-than-operator.english.md:
````diff
@@ -9,15 +9,7 @@ videoUrl: 'https://scrimba.com/c/cNVRWtB'
-'8' < 4  // false
-```
-
+<blockquote>2 &lt; 5  // true<br>'3' &lt; 7  // true<br>5 &lt; 5  // false<br>3 &lt; 2  // false<br>'8' &lt; 4  // false</blockquote>
````
Shouldn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-less-than-or-equal-to-operator.english.md:
````diff
@@ -9,15 +9,7 @@ videoUrl: 'https://scrimba.com/c/cNVR7Am'
-'8' <= 4  // false
-```
-
+<blockquote>4 &lt;= 5  // true<br>'7' &lt;= 7  // true<br>5 &lt;= 5  // true<br>3 &lt;= 2  // false<br>'8' &lt;= 4  // false</blockquote>
````
Shoudn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-strict-equality-operator.english.md:
````diff
@@ -10,12 +10,7 @@ videoUrl: 'https://scrimba.com/c/cy87atr'
-3 === '3' // false
-```
-
+<blockquote>3 === 3   // true<br>3 === '3' // false</blockquote>
````
Shouldn't this be js?

ojeytonwilliams commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/comparison-with-the-strict-inequality-operator.english.md:
````diff
@@ -9,13 +9,7 @@ videoUrl: 'https://scrimba.com/c/cKekkUy'
-4 !== 3   // true
-```
-
+<blockquote>3 !== 3   // false<br>3 !== '3' // true<br>4 !== 3   // true</blockquote>
````
Shouldn't this be js?

ojeytonwilliams reviewed (changes requested):
There were a few cases where I thought some reverted blockquotes should have been kept as js.  I've checked everything locally, though, and it looks great otherwise!

RandellDawson commented:
@ojeytonwilliams Regarding the several challenges you asked if they should be `j`, I originally I had them as code blocks, but @moT01 did a first pass (informal) review yesterday and thought they should be reverted back to blockquotes.  Looking at it again, I can see how they technically are JavaScript expressions.

RandellDawson committed:
fix: change js to json

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

moT01 commented:
There was a misunderstanding when I pointed out those comparison challenges yesterday @RandellDawson - I wasn't saying that they weren't `js` - what I meant was that they were misaligned and looked goofy

ojeytonwilliams commented:
Fair enough.  @moT01 do you still prefer them as blockquotes?  I just figured they might as well be treated as code.

moT01 commented:
I think they work either way. I'll leave it up to Randell - if he wants to leave them, sure - or if he want's them changed back, he'll be the one to do it

RandellDawson committed:
fix: convert various section to triple backticks

RandellDawson commented:
@ojeytonwilliams OK.  Hopefully this commit takes care of everything.  I made the 10 changes suggested above.

RandellDawson commented:
@moT01 If it is all the same to you (which seems like it is), I have reverted back to the code blocks, since they really were JavaScript expressions.   I will also fixed the merge conflict on this, so we can get it merged soon.

ojeytonwilliams commented:
I'm just checking this one final time.  Before I merge, there's potential conflict with three of your PRs, @moT01.  Shall we just merge this big one and deal with any conflicts in the smaller PRs?  Seems easier than repeatedly changing this one.  What do you think?

moT01 commented:
yes, that's fine

ojeytonwilliams committed:
fix: Make the formatting consistent for comparisons

ojeytonwilliams commented:
Okay, cool.  I went ahead and made some formatting changes.  I figured those were what you were talking about in the first place, @moT01. I've checked they all look fine locally, so if you're happy, please feel free to merge!

moT01 reviewed (approved):
🎉 🎉 🎉 🎉 🎉 🎉 🌮
